docs/docs/migrate-from-v1/README.md

```diff
@@ -0,0 +1,259 @@
+}
+```
+
+Now, in v2 workflows you can `import` your packages in the top of the step, just like a normal Node.js module:
+
+```javascript
+import axios from 'axios';
+
+defineComponent({
+  async run({ steps, $ }) {
+    // your code can be entered here
+  }
+});
+```
+
+Allowing all of the scaffolding to be edited opens up the ability to [pass props](code/nodejs/#passing-props-to-code-steps) into your Node.js code steps, which we'll cover later.
+
+### Step Exports 
+
+In v1, you could assign arbitrary properties to `this` within a Node.js step and the properties would be available as step exports:
+
+```javascript
+// this step's name is get_customer_data
+async (event, steps) {
+  this.name = 'Dylan';
+  // downstream steps could use steps.get_customer_data.name to retrieve 'Dylan'
+}
+```
+
+In v2, this specific way of exporting data is now available with `$.export`:
+
+```javascript
+// this step's name is get_customer_data
+defineComponent({
+  async run({ steps, $ }) {
+    $.export('name', 'Dylan');
+    // downstream steps can use steps.get_customer_data.name to retrieve 'Dylan'
+  }
+});
+```
+::: tip
+Using `return` to export data is the same from v1 to v2. You can still `return` data, and it will be available to other steps with `steps.[stepName].$return_value.
+:::
+
+### Exiting a workflow early 
+
+In v1, the `$exit` function can be called to exit a flow early:
+
+```javascript
+async (event, steps) {
+  exit('Exiting the whole workflow early');
+  console.log('I will never run');
+}
+```
+
+In v2, this same function is available, but under `$.exit`:
+
+```javascript
+defineComponent({
+  async run({ steps, $ }) {
+    $.flow.exit('Exiting the workflow early');
+    console.log('I will never run');
+  }
+});
+```
+
+::: warning
+In v2, `$end` does not display the reason in the workflow results logs.
+::: 
+
+### Params vs Props
+
+In workflow builder v2, both Component **props** and Node.js code steps **props** are one in the same.
+
+You can still enter free text and select data from other steps in pre-built actions. Also can add your own custom props that accept input like strings, numbers and more just like in v1.
+
+#### Defining params
+
+In the v1 workflow builder, params could be structured or unstructured. The [params schema builder](https://pipedream.com/docs/v1/workflows/steps/params/#configuring-custom-params) allowed you to add your own custom params to steps.
+
+In v2, you can add your own custom props without leaving the code editor with `props`.
+
+```javascript
+export default defineComponent({
+  props: {
+    firstName: {
+      type: 'string',
+      label: 'Your first name',
+    }
+  },
+  async run({ steps, $ }) {
+    console.log(this.firstName);
+  }
+});
+
+```
+
+In the example above a `firstName` string prop is created. The value assigned to this prop in the workflow builder by either a static string or dynamic data will be available in the step as `this.firstName`.
```

Comment:
"In the example, you added a `firstName` string prop. "

I would also shorten this sentence, it was hard for me to follow: "The value assigned to this prop in the workflow builder by either a static string or dynamic data will be available in the step as `this.firstName`."